### DISCUSSION TITLE
Multiplatform Docker images for Quarto

### DISCUSSION THREAD

**Discussion by jdutant** [quarto-dev/quarto-cli#5129] [Show and Tell]
Hi all,

There are quite a few docker images for Quarto around, but I haven't seen any that works on new Macs (Apple Silicon, aka aarch64 or arm64) as well as amd64 (see https://github.com/quarto-dev/quarto-cli/discussions/3308#discussioncomment-4157022). So here are a couple:

1. Minimal: just quarto-cli. 

    💾 `docker pull jdutant/quarto-minimal`

2. Latex: a basic Tex Live installation (not TinyTeX because Quarto can't install it on arm64)

   💾 `docker pull jdutant/quarto-latex`

Try them out (this will pull them from docker hub):

``` bash
docker run --rm --volume $(pwd):/data jdutant/quarto-minimal -v
docker run --rm --volume $(pwd):/data jdutant/quarto-latex -v
```

See [the repository](https://github.com/jdutant/quarto-dockerfiles) to customize and build them yourself. It also illustrates how to use them as GitHub actions.

 [jdutant/quarto-dockerfiles](https://github.com/jdutant/quarto-dockerfiles) 

Image stages are completely modular (each does its own `apt-get` update), so it should be relatively easy to graft the steps that install Quarto and LaTeX onto a multiplatform base image for R or Python. 

Currently they run Quarto 1.3.313 on ubuntu (I couldn't use alpine as Quarto needs glibc). But you can build them with another version using `--build-arg QUARTO_VERSION=1.3.306`. Note that older versions like 1.2.475 don't have a arm64 build, so they would only work on amd64 machines. 

Enjoy 🚀 

PS I'd like to add instructions on using docker-compose and a volume to persist any LaTeX packages installed on the fly but haven't gone round to do it. If you do that please let me know!

**Comment by jimjam-slam** [comment]
[`rocker-r-ver`](https://github.com/rocker-org/rocker-versioned2/releases) supports aarch64 and has a Quarto install script 🙂  I've only used it through a development container (I'm actually writing a blog post this weekend), but I assume it works fine as a raw Docker image too! Performance on my Apple Silicon Macbook has improved significantly since I switched to a native build.
